example/example.md

```diff
@@ -0,0 +1,601 @@
+# Ably-Flutter examples
+
+## Authentication
+
+### Authenticate using [key authentication](https://ably.com/docs/core-features/authentication/#basic-authentication)
+
+```dart
+// Create an instance of ClientOptions with Ably key
+final clientOptions = ably.ClientOptions(key: 'xxxx:xxxx');
+
+// Use ClientOptions to create realtime or rest instance
+ably.Realtime realtime = ably.Realtime(options: clientOptions);
+ably.Rest rest = ably.Rest(options: clientOptions);
+```
+
+### Authenticate using [token authentication](https://ably.com/docs/core-features/authentication/#token-authentication)
+
+```dart
+// Create an instance of ClientOptions with Ably token
+ably.ClientOptions clientOptions = ably.ClientOptions(
+    key: 'xxxxxxxx', 
```

Comment:
Yes, the same as in `ably-java` the key and token are differentiated by the `:` character 🙂

Reply:
Okay, so as far as i understand, from here: https://ably.com/docs/core-features/authentication/#token-authentication it's enough that token is provided to use the token auth. Under the hood, if key contains `:` it's used as a `key`, but if it doesn't it's used as a `token`. Auth callback is not necessary, but there's this part in the docs:

`The last of those (providing a literal token or tokenDetails) is mostly only used for testing: since tokens are short-lived, in production you almost always want to use an authentication method that allows the client library to renew the token automatically when the previous one expires.`

So I guess that's why the authCallback is always visible in the examples

Reply:
So to me, it makes sense to split this into two code examples: static token auth (providing token via the `key`) and dynamic token auth (providing token via the `authCallback`). But on the other hand, if all other code examples in Ably show them together in a single code snippet, then I'm fine with leaving it as it is :wink: I'll leave the decision to you :grin: